core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -1739,4 +1739,61 @@ public void testSqlForInvalidArguments() {
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Invalid dialect: (empty string)");
   }
+
+  @Test
+  public void testReplaceViewWithUpdatedColumnDoc() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withDefaultCatalog(catalog().name())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    view.replaceVersion()
+        .withDefaultCatalog(view.currentVersion().defaultCatalog())
+        .withDefaultNamespace(identifier.namespace())
+        .withSchema(view.schema().withColumnDoc("data", "some docs for data"))
+        .withQuery("spark", view.sqlFor("spark").sql())
+        .commit();
```

Comment:
We can actually take this a step further by adding an API to view , view.updateColumnDoc which returns `ReplaceViewVersion` which is already initialized with namespace/catalog/query details since those will all be the same for someone who just wants to update the doc.

Reply:
Actually I think that's more than just a nice to have, since we need to make sure all dialects are preserved in the replace. It's good to have that be in the API implementation itself rather than have different engine behaviors.